spec/controllers/challenge_claims_controller_spec.rb

```diff
@@ -25,6 +25,28 @@
       it_redirects_to_with_error(root_path, \
                                  "You aren't allowed to see that user's claims.")
     end
+
+    context "for a prompt meme" do
+      let(:signup) { create(:prompt_meme_signup) }
+
+      it "will not allow a logged in user to see everyone's claims" do
+        fake_login_known_user(user)
+
+        get :index, params: { collection_id: collection.name }
+
+        it_redirects_to_with_error(collection, "Sorry, you're not allowed to do that.")
+      end
+
+      it "will allow a maintainer to see everyone's claims" do
+        collection.collection_participants.create(pseud: user.pseuds.first, participant_role: "Moderator")
+        fake_login_known_user(user)
+
+        get :index, params: { collection_id: collection.name }
+
+        expect(flash[:error]).to be_blank
+        expect(assigns(:claims))
```

Comment:
Yep, I welcome ideas for what to test there. This spec seems to be centred on testing redirects, so I didn't have a pattern for what positive things to look for. I added a spec for the happy path because I had accidentally broken it at some point, so regressions are always possible, and well, many people recommend testing primarily for the happy path. 🤷‍♀️

Reply:
I think what would probably be most thorough here would be to create a claim in a different collection: 
```
let(:claim_in_other_collection) { create(:challenge_claim) }
```

Then you could check that the right one is included and the wrong one isn't:
```
expect(assigns(:claims)).to include(claim)
expect(assigns(:claims))not_.to include(claim_in_other_collection)
```
If you wanted to be _really_ thorough, you could create an additional claim in this collection and assign it to a specific `claiming_user_id`, ensuring that you see claims by more than one user on this page.

Reply:
Speaking of that: I only used `assigns` because it was done in another test (codemonkey see, codemonkey do), and I wasn't sure what would be the most relevant test at that time.

Now you've given me a few scenarios (thank you!), should we create more tests using `assigns`, if it's going to be removed from Rails? Do we want to keep the gem that lets them work in future Rails versions, or do we want to phase them out and not create any new ones?

Reply:
I think there's definite value to the tests and it's worth keeping the gem if there's no new equivalent being introduced. However, I have no idea if there is/is not an equivalent coming, so I'll defer to @redsummernight here.